landingai/predict.py

```diff
@@ -191,14 +212,39 @@ def serialize_rois(rois: List[List[Tuple[int, int]]], mode: str) -> str:
 
 
 class EdgePredictor(Predictor):
+    """`EdgePredictor` runs local inference by connecting to an edge inference service (e.g. LandingEdge)"""
+
     def __init__(
-        self,
-        host: str,
-        port: int,
+        self, host: str = "localhost", port: int = 8000, check_server_ready: bool = True
     ) -> None:
+        """By default the inference service runs on `localhost:8000`
+
+        Parameters
+        ----------
+        host : str, optional
+            Hostname or IP, by default "localhost"
+        port : int, optional
+            Port, by default 8000
+        check_server_ready : bool, optional
+            Check if the inference server is running, by default True
+        """
         self._url = f"http://{host}:{port}/images"
+        # Check if the inference server is reachable
+        if check_server_ready:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(1)
+            result = sock.connect_ex((host, port))
+            if result != 0:
+                raise ConnectionError(
+                    f"Failed to connect to the model server. Please check if the server is running and the connection url ({self._url})."
+                )
+            sock.close()
```

Comment:
how about abstract it out as a function and reuse it for both predictors?
just for keeping it DRY

Reply:
Done.